lib/portlayer/exec/container_cache.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"sync"
+)
+
+/*
+* ContainerCache will provide an in-memory cache of containerVMs.  It will
+* be refreshed on portlayer start and updated via container lifecycle
+* operations (start, stop, rm) and well as in response to infrastructure
+* events
+ */
+type containerCache struct {
+	m sync.RWMutex
+
+	// cache maps containerVM to container ID && infrastructure ID
+	cacheByID      map[string]*Container
```

Comment:
@hmahmood "next to nil chance of a clash between id and reference string". It's unlikely, but getting ill-defined or incorrect behavior if you name a container "vm-15" is a hole we don't need to intentionally open up.

@cgtexmex Wasn't requesting you change the code, rather just wanting to understand the choice behind the design. The point Hasan raises may be the very reason to have the two hashtables.

I would like to make sure that if someone does choose to name a container the same string as an existing moref - however unlikely - that we don't return the wrong container. In some respects, the double hashtable implementation makes it easier to avoid that kind of clash.

@cgtexmex can you please add a unit test to verify that naming a container the same as an existing moref gives the expected result. Thx